src/main/java/io/confluent/idesidecar/websocket/resources/WebsocketEndpoint.java

```diff
@@ -0,0 +1,326 @@
+        headers.messageType(),
+        workspaceSession.processId()
+    );
+
+    /*
+      At this time, all websocket messages received from workspaces are intended to
+      be proxied to all of the other workspaces. Do so here.
+     */
+
+    int otherCount = sessions.size() - 1;
+    if (otherCount <= 0) {
+      Log.debug("No other workspaces to broadcast message to.");
+      return;
+    } else {
+      Log.debugf(
+          "Proxying message %s from workspace %d to %d other workspace(s)",
+          headers.id(),
+          workspaceSession.processId(),
+          otherCount
+      );
+      sessions.entrySet().stream()
+          .filter(pair -> pair.getValue().processId() != workspaceSession.processId() && pair.getKey().isOpen())
+          .forEach(pair -> {
+            pair.getKey().getAsyncRemote().sendText(messageString);
+          });
+    }
+  }
+
+  @OnError
+  public void onError(Session session, Throwable throwable) {
+    // May or may not actually remove -- if had not yet been authorized, it won't be in the map.
+    // (but will definitely not be in the map after this statement.)
+    WorkspaceSession existingSession = sessions.remove(session);
+
+    var id = existingSession != null ? existingSession : session.getId();
+    String logMsg = "Websocket error for authorized workspace: %s - %s";
+    Log.errorf(logMsg, id, throwable.getMessage());
+    Log.errorf(logMsg, id, "Session removed.");
+  }
+
+  @OnClose
+  public void onClose(Session session) {
+    WorkspaceSession existing = sessions.remove(session);
+    Log.info("Websocket session closed: " + session.getId());
+
+    if (existing != null) {
+      // was a registered workspace session. Announce to all other workspaces that the list has changed.
+      Log.infof(
+          "Closed session was workspace %d, broadcasting workspace count change.",
+          existing.processId()
+      );
+      try {
+        broadcastWorkspacesChanged();
+      } catch (java.io.IOException e) {
+        Log.errorf("Failed to broadcast workspace removed message: %s", e.getMessage());
+      }
+    }
+  }
+
+  /**
+   * Parse and validate a message from a websocket session. If the message is not parseable or
+   * fails coarse validation,IOException is raised
+   * @param message JSON spelling of a {@link io.confluent.idesidecar.websocket.messages.Message} object
+   * @return The parsed message
+   * @throws java.io.IOException on any error.
+   */
+  @NotNull
+  private Message parseAndValidateMessage(String message) throws java.io.IOException {
+    Message m = mapper.readValue(message, Message.class);
+
+    MessageHeaders headers = m.headers();
+
+    // header.originator for messages recv'd by sidecar must always be a string'd integer
+    // representing the workspace id (process id).
+    if (!headers.originator().matches("\\d+")) {
+      throw new IOException(
+          "Invalid websocket message header originator value: " + headers.originator()
+      );
+    }
```

Comment:
Strictly speaking, a large sequence of digits would successfully match this regex, but then fail when we try to parse it as a long. Should we just try to parse it as a long here, or if we want to use a regex limit the number of digits?
```suggestion
    if (!headers.originator().matches("\\d{1,18}")) {
      throw new IOException(
          "Invalid websocket message header originator value: " + headers.originator()
      );
    }
```
The max `long` value is 2^23-1, which is 19 digits long: `9,223,372,036,854,775,807`. But not all 19-digit integral numbers fits within `long`.

Reply:
Refactored to no longer be needed. Now just the `parseInt()` within `onMessage()` handles it all. This `parseAndValidate..` used to do a lot more. Now does nothing so has evaporated.